include/ck_tile/ops/reduce/kernel/multi_reduce2d_multiblock_kernel.hpp

```diff
@@ -0,0 +1,314 @@
+
+                block_reduce2d(x_compute, y_compute, reduce_ops.get(number<i>{}));
+
+                move_tile_window(x_window, {0, S::Block_N});
+            }
+
+            block_reduce2d_sync(y_compute, reduce_ops.get(number<i>{}));
+            block_reduce2d_cross_warp_sync(
+                y_compute, static_cast<void*>(smem), reduce_ops.get(number<i>{}));
+
+            if(thread_n_cluster_id == 0)
+            {
+                tile_elementwise_inout(accumulator_ops.get(number<i>{}), y_compute, y_compute);
+
+                // 1. Get the pointer to the output buffer for this tile window
+                auto* p_y_tile = p_y_tuple + (i * output_tensor_offset) +
+                                 S::Block_M * block_group_id +
+                                 y_compute.get_thread_buffer_size() * thread_m_cluster_id;
+
+                // 2. Cast y_compute to thread_buffer<YDataType, N> (N = tile size)
+                auto y_thread_buf = cast_tile<YDataType>(y_compute).get_thread_buffer();
+
+                // 3. Atomically operation between the register tile and DRAM
+                auto atomic_ops =
+                    interblock_reduce_ops.get(number<i>{})
+                        .template GetAtomic<YDataType, y_thread_buf.N>(); // TODO: check if we
+                                                                          // need YDataType
+                atomic_ops(p_y_tile, y_thread_buf);
+            }
+        });
+    }
+
+    /// @brief Validates if the given arguments are supported by the 2D multi reduction kernel.
+    ///
+    /// @param y_continous_dim Size of the continuous dimension of the output tensor.
+    ///                        Must be a multiple of ThreadTile_N for proper thread mapping.
+    ///
+    /// @param input_strides   The stride configuration of the input tensor.
+    ///                        The last stride must be 1 to ensure contiguous memory access
+    ///                        and enable efficient vectorized loads.
+    ///
+    /// @return true if the arguments are supported, false otherwise.
+    ///         Error messages are logged when CK_TILE_LOGGING is enabled.
+    ///
+    /// @note Requirements:
+    ///       - y_continous_dim % ThreadTile_N == 0 (for proper thread distribution)
+    ///       - input_strides[-1] == 1 (for contiguous memory access)
+    template <typename InputStrides>
+    CK_TILE_HOST static bool IsSupportedArgument(index_t y_continous_dim,
+                                                 InputStrides input_strides)
+    {
+        using S = typename Problem::BlockShape;
+
+        if(y_continous_dim % S::ThreadTile_N != 0)
+        {
+            if(ck_tile::EnvIsEnabled(CK_TILE_ENV(CK_TILE_LOGGING)))
+            {
+                CK_TILE_ERROR("Total reduction size should be a multiple of ThreadTile_N!");
+            }
+            return false;
+        }
+
+        if(input_strides.at(number<input_strides.size() - 1>{}) != 1)
```

Comment:
Why's that? You don't have to do vectorized reads only on rightmost (innermost) dimension .